src/main/kotlin/com/bugsnag/android/gradle/internal/BugsnagHttpClientHelper.kt

```diff
@@ -40,14 +69,38 @@ class LegacyBugsnagHttpClientHelper(
     override val okHttpClient: OkHttpClient by lazy { newClient(timeoutMillis.get()) }
 }
 
+internal class ProxyAuthenticator(val user: String, val pass: String) : java.net.Authenticator() {
+    override fun getPasswordAuthentication() = PasswordAuthentication(user, pass.toCharArray())
+}
+
 internal fun newClient(timeoutMillis: Long): OkHttpClient {
     val timeoutDuration = Duration.ofMillis(timeoutMillis)
-    return OkHttpClient.Builder()
+    val builder = OkHttpClient.Builder()
         .readTimeout(timeoutDuration)
         .writeTimeout(timeoutDuration)
         .connectTimeout(timeoutDuration)
         .callTimeout(Duration.ZERO)
-        .build()
+    configureHttpProxySettings(builder)
+    return builder.build()
+}
+
+private fun configureHttpProxySettings(builder: OkHttpClient.Builder) {
+    val host: String? = System.getProperty("http.proxyHost")
+    val port: String? = System.getProperty("http.proxyPort")
+
+    if (host != null && port != null) { // set the proxy host/port if set
+        val socketAddress = InetSocketAddress("localhost", port.toInt())
+        val proxy = Proxy(Proxy.Type.HTTP, socketAddress)
+        builder.proxy(proxy)
+    }
+
+    val user: String? = System.getProperty("http.proxyUser")
+    val pass: String? = System.getProperty("http.proxyPassword")
+
+    if (user != null && pass != null) { // set the default authenticator if credentials set
+        java.net.Authenticator.setDefault(ProxyAuthenticator("user", "password"))
```

Comment:
This works for the test but I'm not sure it will in practice 😀 

```suggestion
        java.net.Authenticator.setDefault(ProxyAuthenticator(user, password))
```

Reply:
Whoops :D good catch!